### PR TITLE
Fix for #20716: "Our analytics" can be confusing when user does not have access to that collection

### LINKED ISSUE #20716: "Our analytics" can be confusing when user does not have access to that collection
**Describe the bug**
"Our analytics" can be confusing when user does not have access to that collection.

**To Reproduce**
1. Admin > Permissions > Collections - set "Our analytics" to `No access` for All Users
2. Login as user
3. The homepage should probably not show the `Your teams’ most important dashboards go here` (this is new in 0.42) and perhaps not show the text `Access dashboards, questions, and collections in "Our analytics"`
image
4. Click "Browse all items" - that will take you to collections hierarchy page, but it should probably hide the "Our analytics" in the sidebar and auto-select the first available collection by default:
image
Example with a few collections available to the user:
image

**Information about your Metabase Installation:**
Tested 0.41.6 thru 0.42.1

**Additional context**
Related to #19070 and #8747

### PR DESCRIPTION
**Pull request by calherries**
Fixes https://github.com/metabase/metabase/issues/20716

Basically, "Our analytics" should not be visible from the side menu if the user doesn't have read perms for it.

### Implementation details

Currently a GET request to `/api/collection/root` from a user without root permissions will return a 200 response with details about the root collection. This breaks the FEs expectation that it should only be able to load collections for which it has permissions to read. Any collections requested with `/api/collection/:id` receive a 403 response if the user doesn't have read permissions. Additionally, the FE expects all `/api/collection/root/items` to only contain collections it has permission to read.

So, this change does two things on the BE:
- makes a GET request to `/api/collection/root` return a 403 response if the user doesn't have root permissions
- makes a GET request to `/api/collection/root/items` exclude the root collection if the user doesn't have root permissions

### CODE REVIEW

**Comment by gusaiani** [comment]
@calherries, I think I'm missing something about this fix.

Maybe the issue description (in #20716) lacks something, or the testing instructions in this PR could use more details.

To sum it up, what we see in the issue is the old homepage, so it's not clear to me what to look for in our current layout.

**Comment by calherries** [comment]
@gusaiani That's a fair point. The original issue is out of date. Nemanja's last comment on the issue contains the relevant up-to-date details. Basically, Our analytics should not be visible from the side menu if the user doesn't have read perms for it.

**Comment by alxnddr** [comment]
I updated the saved question picker as well

**Comment by calherries** [comment]
I'd like some BE feedback on the changes I made to adjust the tests.

See [this commit](https://github.com/metabase/metabase/pull/23887/commits/72c824296fe112ca9b462947e84ae33b7079fb88#diff-a6b09b9ecb5cfb6fb2118db8aa7e7c6f6483f79b4e53860bcc2c5e2697311922L1164), from line 1164 onwards.

I removed all the tests that check a root collection's ancestors are an empty list. It seemed to duplicate the behaviour covered by [this test](https://github.com/metabase/metabase/blob/20717-our-analytics-with-no-read-permission/test/metabase/api/collection_test.clj#L930). The tests were failing because the GET requests to `api/collection/root` are wrapped with `mt/with-non-admin-groups-no-root-collection-perms`, which is called from inside the macroexpansion of [`with-collection-hierarchy`](https://github.com/metabase/metabase/blob/20717-our-analytics-with-no-read-permission/test/metabase/models/collection_test.clj#L136). Now that the GET requests return a 403 error the tests fail. 

My question is - should I have adjusted these tests instead of removing them entirely? What were they testing that might not be covered now?

**Review by dpsutton — APPROVED** [review]
backend looks good to me. Thanks for the PR description. This particular PR isn't super hard to follow along but its a good habit to have and helps load up context even for straightforward fixes like here.

File: frontend/src/metabase/query_builder/components/saved-question-picker/SavedQuestionList.jsx [hunk 1]
```diff
@@ -43,46 +44,50 @@ function SavedQuestionList({
...
+                databaseId != null
+                  ? schema.tables.filter(table => table.db_id === databaseId)
+                  : schema.tables;
+              return (
+                <React.Fragment>
+                  {tables
+                    .sort((a, b) =>
+                      a.display_name.localeCompare(b.display_name),
+                    )
+                    .map(t => (
```
Comment (gusaiani) [inline frontend/src/metabase/query_builder/components/saved-question-picker/SavedQuestionList.jsx:66]: I see we didn't really want to refactor anything here, we simply put this block inside a `LoadingWrapper`.

At the same time, a couple ideas that could make this block more palatable.

1. Instead of using `t` as the map's iterated item name, to use `table`. 
2. Extracting what's in this map into a function and reducing nesting depth.

**Review by gusaiani — APPROVED** [review]
Looking good to me.

One minor comment inlined.

**Comment by paoliniluis** [comment]
Will this get pushed to 44 or we leave it for 45?

**Comment by nemanjaglumac** [comment]
> Will this get pushed to 44 or we leave it for 45?

https://github.com/metabase/metabase/issues/20716#issuecomment-1185555079

**Comment by calherries** [comment]
@paoliniluis Yes, it should have been backported. Thanks for catching
